### PR TITLE
Feature/at 8844 j&a mrr 2.11.2.a

### PR DESCRIPTION
**Pull request by DGabriel-CCPO**
The system autogenerates text based on the users input in the [previous step](https://ccpo.atlassian.net/browse/AT-8843).  Users have the flexibility to use the system generated text for create their own.

### CODE REVIEW

File: src/steps/02-EvaluationCriteria/MRR/ReviewBarriers.vue [hunk 1]
```diff
@@ -7,19 +7,187 @@
...
+            class="textarea-max-width"
+            id="BarriersTextArea"
+            :value.sync="barriersToOpportunity"
+            :maxChars="2500"
+            :rows="getRowCount"
+            :validateItOnBlur="true"
+            :rules="[
+                $validators.required('Describe the actions proposed to remove barriers to fair' +
+                 ' opportunity..'),
+                $validators.maxLength(2500)
```
Comment (ajaynalamala-ccpo) [inline src/steps/02-EvaluationCriteria/MRR/ReviewBarriers.vue:24]: Need a more specific validation message for max length. This is not captured in the story but captured in FIGMA. I caught this based on a similar feedback I also received :-)
"
$validators.maxLength(
                  2500,
                  'Limit your description to 2,500 characters or less.'
                ), 
"

File: src/steps/02-EvaluationCriteria/MRR/ReviewBarriers.vue [hunk 2]
```diff
@@ -7,19 +7,187 @@
...
+      this.insertDate = storeData.barriers_follow_on_expected_date_awarded||""
+      this.previousJA = storeData.barriers_j_a_prepared_results||""
+
+      if(storeData.barriers_follow_on_requirement === "YES"){
+        this.defaultSuggestion += "To overcome future barriers to competition," +
+           this.agency + " is preparing a fair opportunity competitive" +
+          " follow-on requirement. The follow-on is expected to be" +
+          " completed, solicited, and awarded by " + this.insertDate + "."
+      }
+      if(storeData.barriers_agency_pursuing_training_or_certs === "YES"){
```
Comment (ajaynalamala-ccpo) [inline src/steps/02-EvaluationCriteria/MRR/ReviewBarriers.vue:150]: Inserting this comment here but this applies to all the 3 if conditions you have for compiling "defaultSuggestion". Looking at the story, it looks like the 3 paragraphs of the defaultSuggestion should always be included. Please double check this. Depending on how this goes, you PR may change significantly. I will continue my review once you resolve the 2 comments that I added.

Comment (DGabriel-CCPO) [reply src/steps/02-EvaluationCriteria/MRR/ReviewBarriers.vue:150]: the Figma shows that these are added in depending on the values captured in the previous screen.

**Review by ajaynalamala-ccpo — APPROVED** [review]
Pulled the branch. Tested all the scenarios. Reported couple of findings that got resolved.

**Pull request merged**